src/espLoader.ts

```diff
@@ -0,0 +1,1008 @@
+                } else {
+                    block_timeout = 3000;
+                }*/ // XXX: Partial block inflate seems to be unsupported in Pako. Hardcoding timeout
+          const block_timeout = 5000;
+          if (this.IS_STUB === false) {
+            timeout = block_timeout;
+          }
+          await this.flash_defl_block(block, seq, timeout);
+          if (this.IS_STUB) {
+            timeout = block_timeout;
+          }
+        } else {
+          throw new ESPError("Yet to handle Non Compressed writes");
+        }
+        bytes_sent += block.length;
+        image = image.slice(this.FLASH_WRITE_SIZE, image.length);
+        seq++;
+        if (reportProgress) reportProgress(i, bytes_sent, totalBytes);
+      }
+      if (this.IS_STUB) {
+        await this.read_reg(this.CHIP_DETECT_MAGIC_REG_ADDR, timeout);
+      }
+      d = new Date();
+      const t = d.getTime() - t1;
+      if (compress) {
+        this.log(
+          "Wrote " +
+            uncsize +
+            " bytes (" +
+            bytes_sent +
+            " compressed) at 0x" +
+            address.toString(16) +
+            " in " +
+            t / 1000 +
+            " seconds.",
+        );
+      }
+      if (calculateMD5Hash) {
+        const res = await this.flash_md5sum(address, uncsize);
+        if (new String(res).valueOf() != new String(calcmd5).valueOf()) {
+          this.log("File  md5: " + calcmd5);
+          this.log("Flash md5: " + res);
+          throw new ESPError("MD5 of file does not match data in flash!");
+        } else {
+          this.log("Hash of data verified.");
+        }
+      }
+    }
+    this.log("Leaving...");
+
+    if (this.IS_STUB) {
+      await this.flash_begin(0, 0);
+      if (compress) {
+        await this.flash_defl_finish();
+      } else {
+        await this.flash_finish();
+      }
+    }
+  }
+
+  async flash_id() {
+    this.log("flash_id");
+    const flashid = await this.read_flash_id();
+    this.log("Manufacturer: " + (flashid & 0xff).toString(16));
+    const flid_lowbyte = (flashid >> 16) & 0xff;
+    this.log("Device: " + ((flashid >> 8) & 0xff).toString(16) + flid_lowbyte.toString(16));
+    this.log("Detected flash size: " + this.DETECTED_FLASH_SIZES[flid_lowbyte]);
+  }
+
+  async hard_reset() {
+    this.transport.setRTS(true); // EN->LOW
+    await this._sleep(100);
+    this.transport.setRTS(false);
+  }
+
+  async soft_reset() {
+    if (!this.IS_STUB) {
+      // "run user code" is as close to a soft reset as we can do
+      this.flash_begin(0, 0);
+      this.flash_finish(false);
+    } else if (this.chip.CHIP_NAME != "ESP8266") {
+      throw new ESPError("Soft resetting is currently only supported on ESP8266");
+    } else {
+      // running user code from stub loader requires some hacks
+      // in the stub loader
+      this.command(this.ESP_RUN_USER_CODE, undefined, undefined, false);
+    }
+  }
+}
+
+export default ESPLoader;
```

Comment:
Let's try to avoid default exports. Named exports are easier to manage and easier to track as it's always same name.